Framework/NexusGeometry/src/InstrumentGeometryAbstraction.cpp

```diff
@@ -0,0 +1,70 @@
+//--------------------
+// Includes
+//--------------------
+
+#include "MantidGeometry/Instrument.h"
+#include "MantidGeometry/Instrument/ReferenceFrame.h"
+#include "MantidGeometry/Instrument/Detector.h"
+#include "MantidNexusGeometry/InstrumentGeometryAbstraction.h"
+#include "MantidNexusGeometry/ShapeGeometryAbstraction.h"
+
+#include <boost/make_shared.hpp>
+
+namespace Mantid{
+namespace NexusGeometry{
+
+///Constructor
+InstrumentGeometryAbstraction::InstrumentGeometryAbstraction(const std::string &instrumentName){
+    Geometry::Instrument_sptr inst_sptr(new Geometry::Instrument(instrumentName));
+    this->instrument_sptr = inst_sptr;
+
+    //Default view
+    std::string defaultViewAxis = "z";
+    Geometry::PointingAlong pointingUp(Geometry::Y), alongBeam(Geometry::Z), thetaSign(Geometry::X);
```

Comment:
I'm not sure about hard-coding reference frame. It depends on what the Nexus standard does about this.

Reply:
This is the NeXus reference frame, unlike IDFs the NeXus standard doesn't allow you to define another reference frame.